mysql/toddl.go

```diff
@@ -228,3 +232,31 @@ func cvtForeignKeys(conv *internal.Conv, srcTable string, srcKeys []schema.Forei
 	}
 	return spKeys
 }
+
+func cvtIndexes(conv *internal.Conv, srcTable string, srcIndexes []schema.Index, schemaIndexKeys map[string]bool) []ddl.CreateIndex {
+	var spIndexes []ddl.CreateIndex
+	for _, srcIndex := range srcIndexes {
+		var spKeys []ddl.IndexKey
+		for _, k := range srcIndex.Keys {
+			spCol, err := internal.GetSpannerCol(conv, srcTable, k.Column, true)
+			if err != nil {
+				conv.Unexpected(fmt.Sprintf("Can't map index for table %s", srcTable))
+				continue
+			}
+			spKeys = append(spKeys, ddl.IndexKey{Col: spCol, Desc: k.Desc})
+		}
+		if srcIndex.Name == "" {
+			b := make([]byte, 4)
+			_, err := rand.Read(b)
+			if err != nil {
+				conv.Unexpected(fmt.Sprintf("Can't map index for table %s", srcTable))
+				continue
+			}
+			srcIndex.Name = fmt.Sprintf("Index_%s_%x-%x", srcTable, b[0:2], b[2:4])
+		}
+
+		spKeyName := internal.GetSpannerIndexKeyName(srcIndex.Name, schemaIndexKeys)
+		spIndexes = append(spIndexes, ddl.CreateIndex{Name: spKeyName, Unique: srcIndex.Unique, Keys: spKeys})
+	}
+	return spIndexes
+}
```

Comment:
this function seems to be duplicated between mysql and postgresql - one way to remove this duplication is to define a function on schema.Index to produce the corresponding ddl.CreateIndex statement since a given schema.Index struct contains all (but srcTable) fields required to produce the corresponding ddl.CreateIndex struct

Reply:
if we make this a function on schema.Index, we need to put this in schema package, but in that case we will get cyclic import issue, as schema will be importing from internal and internal is importing from schema.

Reply:
I see, we will probably get to it as part of larger refactoring. Sounds good to keep at as it is. Resolving.